site/docs/v1/tech/guides/user-actions.adoc

```diff
@@ -0,0 +1,874 @@
+
+==== Create Survey Action With Options With Localizations
+
+In this last Action, you will add Options that represent the responses a user may have in the survey. You will also add a translation (localization) for each Option so that agents who don't speak English can see feedback in their own language.
+
+* Navigate to [breadcrumb]#Settings -> User Actions# and add icon:plus[role=ui-button green,type=fas] a new User Action.
+** [field]#Id#: `8e6d80df-74bb-4cb8-9caa-c9a2dafc6e57`
+** [field]#Name#: `Survey`
+** Leave all temporal, email, and notification settings disabled.
+** Under the [breadcrumb]#Options# tab, click [uielement]#Add option# to add the first option.
+*** [field]#Name#: `Good`
+*** Click [uielement]#Add localization#.
+**** [field]#Locale#: `Esperanto`
+**** [field]#Text#: `Bona`
+*** Click [uielement]#Submit# to save the option.
+** Add a second option by clicking the [uielement]#Add option# button.
+*** [field]#Name#: `Neutral`
+*** Click [uielement]#Add localization#.
+**** [field]#Locale#: `Esperanto`
+**** [field]#Text#: `Meza`
+*** Click [uielement]#Submit# to save the option.
+** Add a third option by clicking the [uielement]#Add option# button.
+*** [field]#Name#: `Bad`
+*** Click [uielement]#Add localization#.
+**** [field]#Locale#: `Esperanto`
+**** [field]#Text#: `Malbona`
+*** Click [uielement]#Submit# to save the option.
+*  Save icon:save[role=ui-button blue,type=fas] the User Action.
+
+image::guides/user-actions/webhook-options.png[Create a Webhook With Options in FusionAuth,width=1200,role=bottom-cropped]
+
+== PiedPiper Work
+
+Your JavaScript code will act as PiedPiper, Intercom, and Slack, all in one. You'll use the `fusionauth-example-5-minute-guide` Node.js app as the base to start from. If you have not worked through link:/docs/v1/tech/getting-started/5-minute-setup-guide[that guide] and do not have the code available, please do so before continuing.
+
+* Set the `CLIENT_ID` and `CLIENT_SECRET` in your `.env` file to the values you recorded for the new PiedPiper Application in this link:#create-piedpiper-application[section].
+* Create a new environment variable in the `.env` file called `API_KEY` and set the value to the value of the API key you created earlier `FTQkSoanK7ObbNjOoU69WDVclfTx8L_zfEJbdR8M0xu-jKotV0iQZiQh`.
+* Note in the `package.json` file that the `@fusionauth/typescript-client` library is available for use. This is what will be calling the FusionAuth API to create Action instances.
+
+=== Create Mock Intercom API
+
+In the `fusionauth-example-5-minute-guide` Node.js app, open `app.js`.
+
+You'll add a new route that pretends to be Intercom and will listen for new subscribers to start the onboarding process. In this tutorial, the API will just print the webhook to the console so that you can see what it looks like.
+
+At the very top of the file, add a reference to the FusionAuth TypeScript client library, the `API_KEY` and `BASE_URL` environment variables.
```

Comment:
I had a problem here.  I added to the very top of the file, but when the application ran the variables were undefined.  I found I had to move the declaration of the variables after the var dotenv = require('dotenv').config();
```suggestion
After the line `var indexRouter = require('./routes/index');`, add a reference to the FusionAuth TypeScript client library, the `API_KEY` and `BASE_URL` environment variables.
```